### PR TITLE
Build optimized regex from string list

### PR DESCRIPTION
**Pull request by mvorisek**
extracted from https://github.com/doctrine/sql-formatter/pull/119 as requested

Total Tokenizer runtime reduction is about -20%.

See https://3v4l.org/JKn6D/rfc#vgit.master_jit repro repro for the matching speed difference.

### CODE REVIEW

File: src/Tokenizer.php [hunk 1]
```diff
@@ -797,6 +778,68 @@ public function __construct()
         $this->nextTokenRegexNonReserved       = '/\G.*?(?=$|\s|["\'`]|' . $regexBoundaries . ')/';
     }
 
+    /** @param list<string> $values */
+    private function makeRegexFromList(array $values, bool $sorted = false): string
```
Comment (greg0ire) [inline src/Tokenizer.php:789]: The second parameter appears unused.

**Review by greg0ire — COMMENTED** [review]
Such clever tricks should come with an explanation (located in the description of `makeRegexFromList`).

File: src/Tokenizer.php (hunk 1 above)
Comment (mvorisek) [reply src/Tokenizer.php:789]: it is used - it is called recursively

**Comment by mvorisek** [comment]
done - description above the private method added

File: src/Tokenizer.php (hunk 1 above)
Comment (greg0ire) [reply src/Tokenizer.php:789]: Oh right, didn't spot that 👍

**Review by greg0ire — APPROVED** [review]

**Review by SenseException — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @mvorisek !

**Pull request merged**